src/app/file-manager/file-explorer/file-explorer.component.ts

```diff
@@ -68,12 +71,23 @@ export class FileExplorerComponent implements OnInit {
     ];
   }
 
+  countFiles() {
+    this.fileCount = 0;
+    this.selectedFiles.forEach(f => {
```

Comment:
Is there a way to observe `selectedFiles` in angular? If so, you could replace the explicit method calls to `countFIles()` with an observer/subscribe pattern.

Reply:
Since we don't modify selectedFiles directly, it is modified by Primeng's p-treeTable ([(selection)]="selectedFiles") it is not that easy to implement the Observable approach. I suggest to postpone this until all pending PR are merged.